sentry-kotlin-multiplatform/src/androidMain/kotlin/io/sentry/kotlin/multiplatform/SentryInit.android.kt

```diff
@@ -1,15 +1,72 @@
 package io.sentry.kotlin.multiplatform
 
+import android.content.ContentProvider
+import android.content.ContentValues
 import android.content.Context
+import android.database.Cursor
+import android.net.Uri
 import io.sentry.android.core.SentryAndroid
 import io.sentry.kotlin.multiplatform.extensions.toAndroidSentryOptionsCallback
 
-internal actual fun initSentry(context: Context?, configuration: OptionsConfiguration) {
+internal actual fun initSentry(configuration: OptionsConfiguration) {
     val options = SentryOptions()
     configuration.invoke(options)
-    context?.let {
-        SentryAndroid.init(it, options.toAndroidSentryOptionsCallback())
-    }
+    SentryAndroid.init(applicationContext, options.toAndroidSentryOptionsCallback())
 }
 
+internal lateinit var applicationContext: Context
+    private set
+
 public actual typealias Context = Context
+
+/**
+ * A ContentProvider that does NOT store or provide any data for read or write operations.
+ *
+ * It's only purpose is to retrieve and store the application context in an internal top-level
+ * variable [applicationContext]. The context is used for [SentryAndroid.init].
+ *
+ * This does not allow for overriding the abstract query, insert, update, and delete operations
+ * of the [ContentProvider].
+ */
+internal class ContextProvider : ContentProvider() {
```

Comment:
I think the default proguard rules already exclude any `ContentProvider` classes from being obfuscated these days, but maybe worth double checking if the name is kept in a release build.

Reply:
Yup, it's no obfuscated in the release build